core/lib/vlog/src/lib.rs

```diff
@@ -40,13 +61,71 @@ impl FromStr for LogFormat {
     }
 }
 
+// Doesn't define WARN and ERROR, because the highest verbosity of spans is INFO.
+#[derive(Copy, Clone, Debug, Default)]
```

Comment:
This certainly belongs here. If this crate allows enabling and configuring otlp, then the types relevant for configuration should also come from this type.
`zksync_types` is completely unrelated here, so I don't think it should be there.

Reply:
1. Why would `vlog` go away in favor of `vise`? `vise` is a [general-purpose metrics exporter](https://github.com/matter-labs/vise), which may be useful to many people in the same way that `tracing` and its infra is. `vlog` is an umbrella crate that initializes observability stack for our binaries in a way that is tailored to our preference (e.g. not general-purpose). If anything, I'd rather expect `vise` to be initialized through `vlog`, if we want consistency here. These two crates serve very different purposes.
2. Even if we consider this functionality to go to `vise` crate, why wouldn't these types go there? This enum configures the level of events we want to emit. If library configures emitting events, why would a type that defines the minimum level of emitted event live elsewhere? I mean, [`tracing` has one](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/struct.LevelFilter.html), [`log` has one](https://docs.rs/log/latest/log/enum.Level.html), etc. What we don't want this library to know is how we get this value (e.g. from env, from file, etc) -- so that we detach config definition from config (de)serialization, but the config type per se certainly lies in the domain of the library.

Reply:
I think it's important to mention that the aim for this PR is to make us of **traces** that come with Opentelemetry stack. We could also collect metrics via Opentelemetry but I don't see a point in that unless Promtheus-style polling stops suiting our needs for some reason. So in that sense `vise` and whatever is introduced in this PR are complementing each other IMHO.